backend/remote-state/kubernetes/client.go

```diff
@@ -0,0 +1,363 @@
+package kubernetes
+
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/md5"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/state"
+	"github.com/hashicorp/terraform/state/remote"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/dynamic"
+)
+
+const (
+	tfstateKey             = "tfstate"
```

Comment:
I've changed these to `tfstate` rather than `terrastate` as that seems to be more consistent with other backends that I've seen.